src/Propel/Generator/Model/ForeignKey.php

```diff
@@ -625,9 +640,8 @@ public function getColumnObjectsMapping()
      */
     public function getMappedForeignColumn($local)
     {
-        $m = $this->getLocalForeignMapping();
-
-        return isset($m[$local]) ? $m[$local] : null;
+        $index = array_search($local, $this->localColumns);
+        return $this->foreignColumns[$index];
```

Comment:
but then we should throw a exception instead of implicitly rely on the warning? This is public api, so it should help the user to find the cause of errors